python/test/test_source.py

```diff
@@ -0,0 +1,118 @@
+from timeplus import Source, Stream
+
+
+def test_source_generator(test_environment):
+    stream_name = "iot"
+    source_name = "iot"
+    # create a new stream
+    stream = Stream(env=test_environment).name(stream_name)
+
+    try:
```

Comment:
one improvement here is that you can call stream.exist() to check if the stream already exist, if yes, delete it and recreate.